src/views/InteractiveMap/CardDialogDesign.vue

```diff
@@ -491,10 +491,15 @@ export default Vue.extend({
           reactionId: modification.id
         });
         // If the objective was set to the added reaction, clear it.
-        this.$store.commit("interactiveMap/setObjectiveReaction", {
-          uuid: this.card.uuid,
-          reaction: null
-        });
+        if (
+          this.card.objective.reaction &&
+          this.card.objective.reaction.id === modification.id
+        ) {
+          this.$store.commit("interactiveMap/setObjectiveReaction", {
```

Comment:
It bothers me a bit that this mutation is different from others. Maybe create `interactiveMap/undoObjectiveReaction` and move `if id == id` inside, for consistency